### PR TITLE
Add functional testing

### PR DESCRIPTION
**Pull request by farquet**
- Spark benchmark setup and teardown has been moved to `SparkUtil` to avoid code duplication.
- maxRetries increased for port allocation in Spark (relates to #66)
- Add `--functional-test` that runs much smaller workload (relates to #96)
- WIP : Add smaller workloads for all benchmarks (ATM `actors` and `apache-spark` benchmark groups are done)

### CODE REVIEW

**Comment by farquet** [comment]
Commit 7ee7375ee77919225e6f09d170f381b55380012b changes `JavaScrabble` to load the resources in a constructor instead of loading them in the static initializer.
The problem with the static initializer is that with the current code, the first iteration will do more work than the subsequent ones, which is conceptually wrong.

**Comment by farquet** [comment]
All benchmarks but 3 have very quick versions now that will be used when `--functional-test` is passed.
Walking through all benchmarks helped find some issues that have been addressed.

Still TODO : 
  - neo4j-analytics : not trivial to create a smaller dataset. (smaller `edges.json` and `vertices`)
  - scala-dotty : requires smaller set of sources to compile. To be discussed.
  - scala-stm-bench7 : not trivial to create a smaller workload.

File: benchmarks/apache-spark/src/main/scala/org/renaissance/apache/spark/SparkUtil.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  }
+
+  def tearDownSparkContext(sc: SparkContext): Unit = {
+    if (sc != null) {
+      sc.stop()
+    }
+  }
+
+  /* For Spark version on renaissance (2.0.0) the Hadoop version is 2.2.0
+   For Windows, the binary zip is not included in the dependencies
```
Comment (axel22) [inline benchmarks/apache-spark/src/main/scala/org/renaissance/apache/spark/SparkUtil.scala]: Minor: please use the

```
/**
 * Asterisk at every line.
 */
```

style, to be consistent.

**Comment by farquet** [comment]
Execution time without `--functional-test` : 
```
real	5m2.101s
user	24m55.034s
sys	2m2.512s
```

Execution time with `--functional-test` : 
```
real	2m57.831s
user	12m6.246s
sys	0m42.209s
```

File: benchmarks/apache-spark/src/main/scala/org/renaissance/apache/spark/SparkUtil.scala (hunk 1 above)
Comment (farquet) [reply benchmarks/apache-spark/src/main/scala/org/renaissance/apache/spark/SparkUtil.scala]: Fixed

**Pull request merged**